app/lib/utils/btcpayserver.js

```diff
@@ -0,0 +1,19 @@
+import get from 'lodash.get'
+
+const parseConnectionString = value => {
+  let config = {}
+  try {
+    config = JSON.parse(value)
+  } catch (e) {
+    return new Error('Invalid connection string')
+  }
+  const configs = get(config, 'configurations', [])
+  const params = configs.find(c => c.type === 'grpc' && c.cryptoCode === 'BTC') || {}
+  const { host, port, macaroon } = params
+  if (!host || !port || !macaroon) {
+    return new Error('Invalid connection string')
```

Comment:
Do we really intend to return an `Error` here rather than `throw` ?

Reply:
Good call. Much better to throw. Updated.